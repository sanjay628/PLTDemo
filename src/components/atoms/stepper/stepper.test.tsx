import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import Stepper from './stepper.screen';

describe('Stepper', () => {
  it('renders initial value and calls onIncQty and onDecQty handlers', () => {
    const onIncQtyMock = jest.fn();
    const onDecQtyMock = jest.fn();
    const {getByText, getByTestId} = render(
      <Stepper value={3} onIncQty={onIncQtyMock} onDecQty={onDecQtyMock} />,
    );
    const valueTextElement = getByText('3');
    expect(valueTextElement).toBeDefined();

    const decreaseButton = getByTestId('decrease-button');
    fireEvent.press(decreaseButton);
    expect(onDecQtyMock).toHaveBeenCalled();

    const increaseButton = getByTestId('increase-button');
    fireEvent.press(increaseButton);
    expect(onIncQtyMock).toHaveBeenCalled();
  });

  it('renders correctly with value 0', () => {
    const {toJSON} = render(<Stepper />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with custom value', () => {
    const {toJSON} = render(<Stepper value={3} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
