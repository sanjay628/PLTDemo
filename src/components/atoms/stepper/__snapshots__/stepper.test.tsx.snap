// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper renders correctly with custom value 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#9191A1",
      "borderRadius": 50,
      "bottom": 16.428571428571427,
      "flexDirection": "row",
      "height": 80,
      "justifyContent": "space-evenly",
      "position": "absolute",
      "right": 15,
      "width": 200,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": 30,
      }
    }
    testID="decrease-button"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 32,
            "fontWeight": "600",
          },
          undefined,
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      -
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontSize": 32,
          "fontWeight": "600",
        },
        undefined,
        Object {
          "color": "#ffffff",
        },
      ]
    }
  >
    3
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": 30,
      }
    }
    testID="increase-button"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 32,
            "fontWeight": "600",
          },
          undefined,
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      +
    </Text>
  </View>
</View>
`;

exports[`Stepper renders correctly with value 0 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#9191A1",
      "borderRadius": 50,
      "bottom": 16.428571428571427,
      "flexDirection": "row",
      "height": 80,
      "justifyContent": "space-evenly",
      "position": "absolute",
      "right": 15,
      "width": 200,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": 30,
      }
    }
    testID="decrease-button"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 32,
            "fontWeight": "600",
          },
          undefined,
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      -
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontSize": 32,
          "fontWeight": "600",
        },
        undefined,
        Object {
          "color": "#ffffff",
        },
      ]
    }
  >
    0
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": 30,
      }
    }
    testID="increase-button"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 32,
            "fontWeight": "600",
          },
          undefined,
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      +
    </Text>
  </View>
</View>
`;
